apiserver/charms.go

```diff
@@ -191,7 +193,15 @@ func (h *charmsHandler) archiveEntrySender(filePath string) bundleContentSenderF
 			io.Copy(w, contents)
 			return nil
 		}
-		return errors.NotFoundf("charm")
+		// Serve the default icon (if the charm icon was requested), or return
+		// a 404 not found response.
```

Comment:
I think this comment would read better inside the if statement:

    if serveDefaultIcon {
        // An icon was requested but none was found in the archive so
        // return the default icon instead.
        ....
    }
    // No file found